packages/backend/src/models/SavedSqlModel.ts

```diff
@@ -0,0 +1,312 @@
+                `${SavedSqlTableName}.last_version_updated_at`,
+                `${SavedSqlTableName}.last_version_updated_by_user_uuid`,
+                `${DashboardsTableName}.name as dashboardName`,
+                `${SavedSqlVersionsTableName}.sql`,
+                `${SavedSqlVersionsTableName}.config`,
+                `${SavedSqlVersionsTableName}.chart_kind`,
+                `${OrganizationTableName}.organization_uuid`,
+                `createdByUser.user_uuid`,
+                `createdByUser.first_name`,
+                `createdByUser.last_name`,
+                `updatedByUser.user_uuid`,
+                `updatedByUser.first_name`,
+                `updatedByUser.last_name`,
+                `${SpaceTableName}.space_uuid`,
+                `${SpaceTableName}.name as spaceName`,
+            ])
+            .where((builder) => {
+                if (options.uuid) {
+                    void builder.where(
+                        `${SavedSqlTableName}.saved_sql_uuid`,
+                        options.uuid,
+                    );
+                }
+
+                if (options.projectUuid) {
+                    void builder.where(
+                        `${ProjectTableName}.project_uuid`,
+                        options.projectUuid,
+                    );
+                }
+
+                // Required filter to join only the latest version
+                void builder.where(
+                    `${SavedSqlVersionsTableName}.created_at`,
+                    '=',
+                    this.database
+                        .from(SavedSqlVersionsTableName)
+                        .max('created_at')
+                        .where(
+                            `${SavedSqlVersionsTableName}.saved_sql_uuid`,
+                            this.database.ref(
+                                `${SavedSqlTableName}.saved_sql_uuid`,
+                            ),
+                        ),
+                );
+            })
+            .orderBy(`${SavedSqlVersionsTableName}.created_at`, 'desc');
+    }
+
+    async get(uuid: string, options: { projectUuid?: string }) {
+        const results = await this.find({ uuid, ...options });
+        const [result] = results;
+        if (!result) {
+            throw new NotFoundError('Saved sql not found');
+        }
+        return SavedSqlModel.convertSelectSavedSql(result);
+    }
+
+    static async createVersion(
+        trx: Knex,
+        data: {
+            savedSqlUuid: string;
+            userUuid: string;
+            config: object;
+            sql: string;
+        },
+    ): Promise<string> {
+        const [{ saved_sql_version_uuid: savedSqlVersionUuid }] = await trx(
+            SavedSqlVersionsTableName,
+        ).insert(
+            {
+                saved_sql_uuid: data.savedSqlUuid,
+                sql: data.sql,
+                config: data.config,
+                chart_kind: ChartKind.VERTICAL_BAR, // todo: get chart kind from config
```

Comment:
I spent longer than I wanna admit thinking I was getting something wrong in the FE cause everything kept getting rendered as bar charts 😂 

That'll still be the case until we update these

Reply:
Fixed in the UI PR as we have the final config types there 👍